### PR TITLE
Multi output support for TrainOnlyTransformerMixin

### LINKED ISSUE #699: [FEATURE] Support multi-output for `TrainOnlyTransformerMixin`
`TrainOnlyTransformerMixin.fit` checks if `y` is 1D. This breaks if you are training multi-label models where `y` is 2D.

To fix this we only need to add `multi_output=True` to `check_X_y(X, y, estimator=self)`.
Line in question:
https://github.com/koaning/scikit-lego/blob/236f4919caf5da736ee5850aabc90850054ae525/sklego/common.py#L84

The only added thing is that `y` can't contain `np.nan` or `np.inf` anymore, which I think is a good thing.

`check_X_y` docs:
https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_X_y.html

I will implement this and make sure all tests still pass.

### PR DESCRIPTION
**Pull request by CarloLepelaars**
# Description

Solves #699 

Adds `multi_output=True` to `check_X_y` in `TrainOnlyTransformerMixin.fit`. 

This allows `TrainOnlyTransformerMixin` to work with multi-output (i.e. 2D) `y` input.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines (ruff)
- [x] New and existing unit tests pass locally with my changes
- [x] All Github Actions pipelines are passing.

### CODE REVIEW

**Review by FBruzzesi — APPROVED** [review]
Thanks @CarloLepelaars for reporting the issue and solving it 🙌🏼👌
